### PR TITLE
Read and push events data from heartbeat

### PR DESCRIPTION
**Pull request by machacekondra**
This PR add a functionality to read the `events` data from `heartbeats`, which contain information which are produced to the specific device object as an k8s event.
```
$ k describe edgedevice c2ebb413-1968-4f81-ac83-a9f99112a26f
....
Events:
  Type     Reason   Age   From                       Message
  ----     ------   ----  ----                       -------
  Warning  Started  33s   edgedeployment-controller  error starting container 1259f0b8025895c7b0cd8dd372eedbb15dfa4d8426231e80b07a85b33cdc2339: cannot listen on the TCP port: listen tcp4 :8888: bind: address already in use,error starting container aae61fab6bb235db942b8ca4173a03e616ab2cad725ea5bacfe197d83b99da8b: a dependency of container aae61fab6bb235db942b8ca4173a03e616ab2cad725ea5bacfe197d83b99da8b failed to start: container state improper
```

```
$ k get events
LAST SEEN   TYPE      REASON    OBJECT                                            MESSAGE
4m18s       Warning   Started   edgedevice/c2ebb413-1968-4f81-ac83-a9f99112a26f   error starting container 1259f0b8025895c7b0cd8dd372eedbb15dfa4d8426231e80b07a85b33cdc2339: cannot listen on the TCP port: listen tcp4 :8888: bind: address already in use,error starting container aae61fab6bb235db942b8ca4173a03e616ab2cad725ea5bacfe197d83b99da8b: a dependency of container aae61fab6bb235db942b8ca4173a03e616ab2cad725ea5bacfe197d83b99da8b failed to start: container state improper

```

### CODE REVIEW

File: internal/yggdrasil/yggdrasil.go [hunk 1]
```diff
@@ -180,6 +184,19 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 			}
 			return operations.NewPostDataMessageForDeviceInternalServerError()
 		}
+
+		// Produce k8s events based on the device-worker events:
+		events := heartbeat.Events
+		for _, event := range events {
+			if event != nil {
```
Comment (eloycoto) [inline internal/yggdrasil/yggdrasil.go:191]: I'll change this a bit to :

```
if event == nil {continue}
```

So it's a bit flat and easy to read.

**Review by eloycoto — COMMENTED** [review]
Minor change, and I'll add test here.

**Review by jakub-dzon — APPROVED** [review]

**Review by jakub-dzon — CHANGES_REQUESTED** [review]
It would be good to have test verifying that events are recorded correctly.

**Comment by machacekondra** [comment]
Added test.

**Review by jakub-dzon — APPROVED** [review]

File: docs/http_api_swagger.md [hunk 2]
```diff
@@ -396,6 +396,23 @@ Status: Internal Server Error
...
+  
+
+
+
+**Properties**
+
+| Name | Type | Go type | Required | Default | Description | Example |
+|------|------|---------|:--------:| ------- |-------------|---------|
+| message | string| `string` |  | | Message describe the event which has occured. |  |
+| reason | string| `string` |  | | Reason is single work description of the subject of the event. |  |
```
Comment (masayag) [inline docs/http_api_swagger.md:411]: s/work/word

the reasons should be predefined in the system, so we can refer the user to documentation about the meaning of each event. 
What are the possibilities for `reason` and can it be documented (in a later stage we'd might be able to recommend on steps to debug those issues)

**Review by pkliczewski — COMMENTED** [review]
/lgtm

File: docs/http_api_swagger.md (hunk 2 above)
Comment (machacekondra) [reply docs/http_api_swagger.md]: Yes, there may be more events generated, where podman produce errors, currently we produce it only for pod start.

**Review by pkliczewski — APPROVED** [review]

**Review by masayag — APPROVED** [review]

**Pull request merged**